src/server/server_state.h

```diff
@@ -206,6 +222,11 @@ class ServerState {  // public struct - to allow initialization.
   mi_heap_t* data_heap_;
   journal::Journal* journal_ = nullptr;
 
+  // Inline scheduling is an optimization for running in low thread count situations.
+  // Since it bypasses the regular transactions mechanism, it has to be disabled in
+  // some situations.
+  bool journal_callbacks_active_ = false;
```

Comment:
Can you move this member to the JournalSlice class and have a fuction to Journal to check if there is journal_callback inside  AllowInlineScheduling